rqd/tests/rqcore_tests.py

```diff
@@ -347,6 +355,182 @@ def test_getRunningFrame(self):
         self.assertEqual(frame, self.rqcore.getRunningFrame(frameId))
         self.assertIsNone(self.rqcore.getRunningFrame('some-unknown-frame-id'))
 
+    @mock.patch.object(rqd.rqcore.RqCore, 'respawn_rqd', autospec=True)
+    def test_restartRqdNowNoFrames(self, respawnMock):
+        self.nimbyMock.return_value.active = False
+
+        self.rqcore.restartRqdNow()
+
+        respawnMock.assert_called_with(self.rqcore)
+
+    @mock.patch.object(rqd.rqcore.RqCore, 'killAllFrame', autospec=True)
+    def test_restartRqdNowWithFrames(self, killAllFrameMock):
+        frame1Id = 'frame1'
+        frame1 = rqd.rqnetwork.RunningFrame(
+            self.rqcore, rqd.compiled_proto.rqd_pb2.RunFrame(frame_id=frame1Id))
+        self.rqcore.storeFrame(frame1Id, frame1)
```

Comment:
Is this polluting the `self.rqcore` object for other tests? We may want to add something to initialize a new `RqCore` object for each test.

Reply:
Nope, it should be fine. `setUp` where it gets created runs before every test. (`setUpClass` is the that runs once before all tests.)

Reply:
Ok, Perfect. Sorry I missed the `setUp` method before.